manager/dispatcher/nodes.go

```diff
@@ -77,6 +80,19 @@ func (s *nodeStore) AddUnknown(n *api.Node, expireFunc func()) error {
 	return nil
 }
 
+// CheckRateLimit returs error if node with specified id is allowed to re-register
+// again.
+func (s *nodeStore) CheckRateLimit(id string) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if existRn, ok := s.nodes[id]; ok {
+		if time.Since(existRn.Registered) < s.rateLimitPeriod {
+			return grpc.Errorf(codes.Aborted, "node with id %s tries to register too fast", id)
```

Comment:
Should this return `codes.Unavailable`? There isn't really a perfect error code for this, so I could see the case for either.

For the text, how about "node %s attempted registration too recently"?